frontend/src/app/shared/models/IModal.ts

```diff
@@ -0,0 +1,9 @@
+export interface IModal {
+    header: string,
+    cancelText?: string,
+    confirmText?: string
+    okText?: string
+    /* eslint-disable  @typescript-eslint/no-explicit-any */
+    component?: any
```

Comment:
Can you use some sort of component class?
Or make something like this https://stackoverflow.com/questions/43933783/pass-component-to-a-component-in-angular